test/e2e/serial/tests/workload_placement.go

```diff
@@ -413,6 +415,305 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
+
+			updatedPod1 = pods[1]
+			By(fmt.Sprintf("checking the pod landed on the target node %q vs %q", updatedPod1.Spec.NodeName, targetNodeName))
+			Expect(updatedPod1.Spec.NodeName).To(Equal(targetNodeName),
+				"node landed on %q instead of on %v", updatedPod1.Spec.NodeName, targetNodeName)
+
+			By(fmt.Sprintf("checking the pod was scheduled with the topology aware scheduler %q", serialconfig.Config.SchedulerName))
+			schedOK, err = nrosched.CheckPODWasScheduledWith(fxt.K8sClient, updatedPod1.Namespace, updatedPod1.Name, serialconfig.Config.SchedulerName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod1.Namespace, updatedPod1.Name, serialconfig.Config.SchedulerName)
+
+			rl1 = e2ereslist.FromGuaranteedPod(updatedPod1)
+			klog.Infof("post-create pod resource list: spec=[%s] updated=[%s]", e2ereslist.ToString(e2ereslist.FromContainers(podSpec.Containers)), e2ereslist.ToString(rl1))
+
+			nrtInitial1, err = e2enrt.FindFromList(nrtInitialList.Items, updatedPod1.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			nrtPostUpdate1, err := e2enrt.FindFromList(nrtPostCreateDeploymentList.Items, updatedPod1.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("checking post-create NRT after pod: %q for target node %q updated correctly", updatedPod1.Name, targetNodeName))
+			dataBefore1, err = yaml.Marshal(nrtInitial1)
+			Expect(err).ToNot(HaveOccurred())
+			dataAfter1, err = yaml.Marshal(nrtPostUpdate1)
+			Expect(err).ToNot(HaveOccurred())
+			match0, err = checkConsumedRes(*nrtInitial1, *nrtPostUpdate1, rl1)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(match0).ToNot(BeEmpty(), "inconsistent accounting: no resources consumed by the running pod,\nNRT before test's pod: %s \nNRT after: %s \n total required resources: %s", dataBefore1, dataAfter1, e2ereslist.ToString(rl1))
+
+			By("deleting the padder pods")
+			// we clean the nodes from the padding pods
+			err = padder.Clean()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("deleting the deployment")
+			err = fxt.Client.Delete(context.TODO(), updatedDp)
+			Expect(err).ToNot(HaveOccurred())
+
+			// the NRT updaters MAY be slow to react for a number of reasons including factors out of our control
+			// (kubelet, runtime). This is a known behaviour. We can only tolerate some delay in reporting on pod removal.
+			Eventually(func() bool {
+				By(fmt.Sprintf("checking the resources are restored as expected on %q", updatedPod0.Spec.NodeName))
+
+				nrtListPostDelete, err := e2enrt.GetUpdated(fxt.Client, nrtPostUpdateDeploymentList, 1*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+
+				nrtPostDelete0, err := e2enrt.FindFromList(nrtListPostDelete.Items, updatedPod0.Spec.NodeName)
+				Expect(err).ToNot(HaveOccurred())
+
+				ok, err := e2enrt.CheckEqualAvailableResources(*nrtPostUpdate0, *nrtPostDelete0)
```

Comment:
don't we want to check equality nrtPostDelete0 vs nrtInitial0? because nrtPostUpdate0 is nrt after applying the deployment update

Reply:
Yep, I think this was a copy paste mistake. Thanks for identifying this.